lib/models/keyword.go

```diff
@@ -0,0 +1,38 @@
+package models
+
+import (
+	"database/sql"
+
+	"github.com/google/uuid"
+	"gorm.io/gorm"
+)
+
+type KeywordStatus string
+
+const (
+	Failed     KeywordStatus = "failed"
+	Pending    KeywordStatus = "pending"
+	Processed  KeywordStatus = "processed"
```

Comment:
As we have `Failed`, should we have `Succeed` or `Completed` instead of `Precessed` (because a failed item was also in `processing` before and so it means it is now also "Processed" – with a failing outcome).